borg/archiver.py

```diff
@@ -1399,7 +1440,7 @@ def build_parser(self, args=None, prog=None):
         subparser.add_argument('location', metavar='ARCHIVE',
                                type=location_validator(archive=True),
                                help='name of archive to create (must be also a valid directory name)')
-        subparser.add_argument('paths', metavar='PATH', nargs='+', type=str,
+        subparser.add_argument('paths', metavar='PATH', nargs='*', type=str,
```

Comment:
> i.e. not releasing it in 1.0.x which isn't too unreasonable given that it's a fairly large and complex new feature (also there is a tangible release plan for 1.1 now)

That's not what I ment. :-)
But what about:
- Release the `--pattern` feature in 1.0.x but leave `nargs="+"` in 1.0.x to remain full commandline compatibility. (So at least one path has to be specified on the commandline.)